File: 2025-11/2025-11-20/longestWord.js
function longestWord(sentence) {

  return sentence
    .replace(/[^a-zA-Z ]/g, "")
    .split(" ")
    .sort((a, b) => b.length - a.length)
    [0];
}

// --- TEST SUITE ---

const testsText = `
1. longestWord("The quick red fox") should return "quick".
2. longestWord("Hello coding challenge.") should return "challenge".
3. longestWord("Do Try This At Home.") should return "This".
4. longestWord("This sentence... has commas, ellipses, and an exclamation point!") should return "exclamation".
5. longestWord("A tie? No way!") should return "tie".
6. longestWord("Wouldn't you like to know.") should return "Wouldnt".
`;

const testsRegex = /(?<number>\d+)\.\s(?<functionCall>.+) should return (?<output>.+)\.$/gm;
const testData = [...testsText.matchAll(testsRegex).map(match => match.groups)];

function runTests(testData) {
  console.log("--------------------------");
  console.log("🧪Starting Verification...");
  console.log("--------------------------");

  let allPassed = true;
  let failCount = 0;
  
  testData.forEach(test => {
    const functionCallOutput = eval(test.functionCall);
    const testOutput = eval(test.output);

    if (functionCallOutput === testOutput) {
      console.log(`${test.number}.✅PASS - Function Call: ${test.functionCall}`);
    }
    else {
      console.log(`${test.number}.❌FAIL - Function Call: ${test.functionCall}\nExpected: ${testOutput}\nGot: ${functionCallOutput}`);
      allPassed = false;
      failCount ++;
    }
  })

  console.log("----------------------------");

  if (allPassed)
    console.log("🎉SUCCESS: All tests PASSED.");
  else
    console.log(`⚠️WARNING: ${failCount}/${testData.length} tests FAILED.`);
}

runTests(testData);

File: 2025-11/2025-11-20/longest_word.py
def longest_word(sentence):
    words = re.sub("[^a-zA-Z ]", "", sentence).split()

    return sorted(words, key = len, reverse = True)[0]

import re

tests_text = '''
1. longest_word("The quick red fox") should return "quick".
2. longest_word("Hello coding challenge.") should return "challenge".
3. longest_word("Do Try This At Home.") should return "This".
4. longest_word("This sentence... has commas, ellipses, and an exclamation point!") should return "exclamation".
5. longest_word("A tie? No way!") should return "tie".
6. longest_word("Wouldn't you like to know.") should return "Wouldnt".
'''

tests_regex = re.compile(r"(?P<number>\d+)\.\s(?P<function_call>.+) should return (?P<output>.+)\.$", re.MULTILINE)
test_data = [item.groupdict() for item in re.finditer(tests_regex, tests_text)]

def run_tests(test_data):
    print("--------------------------")
    print("🧪 Starting Verification...")
    print("--------------------------")

    all_passed = True
    fail_count = 0

    for test in test_data:
        function_call_output = eval(test['function_call'])
        test_output = eval(test['output'])
        
        if function_call_output == test_output:
            print(f"{test['number']}.✅ PASS - Function Call:\n{test['function_call']}")
        else:
            print(f"{test['number']}.❌ FAIL - Function Call:\n{test['function_call']}\nExpected:\n{test_output}\nGot:\n{function_call_output}")
            all_passed = False
            fail_count += 1

    print("----------------------------");

    if (all_passed):
        print("🎉 SUCCESS: All tests PASSED.");
    else:
        print(f"⚠️ WARNING: {fail_count}/{len(test_data)} tests FAILED.");

run_tests(test_data)